File: jmolecules-axon/02-axonstyle-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/event/atm/MoneyDepositedEvent.java
package org.jmolecules.example.axonframework.domain.event.atm;

import org.axonframework.serialization.Revision;
import org.jmolecules.event.annotation.DomainEvent;

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "MoneyDepositedEvent")
public record MoneyDepositedEvent(
  String accountId,
  int amount
) {
  // empty
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/event/atm/MoneyDepositedEvent.kt
package org.jmolecules.example.axonframework.domain.event.atm

import org.axonframework.serialization.Revision
import org.jmolecules.event.annotation.DomainEvent
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.Amount

@Revision("1")
@DomainEvent(namespace = "axon.bank", name = "MoneyDepositedEvent")
data class MoneyDepositedEvent (
    val accountId: AccountId,
    val amount: Amount
)
